src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
+   class Source,Executable daisy1
+   class create,execute dafault_node1
+   
+```
+Let us briefly describe the process:
+1. If a Node is not supported by `Snippets`, then ignore it and proceed to the next one.
+2. If a Node has no `Subgraph`  parents, then replace it with `Subgraph` node and copy the initial Node to the `Subgraph's` body (which is in the `nGraph IR` form).
+3. If a Node has a single `Subgraph` parent, then attach it to the `Subgraph`. It means copy the Node to the `Subgraph's` body, and remove it from the original `nGraph model`. Note that if the Node has more than one parent, corresponding parents' outputs will be connected with the updated `Subgraph` as shown on the diagram below.
+4. If a Node has multiple `Subgraph` parents, then they will be merged into a single `Subgraph` and the Node will be attached to it.
+```mermaid
+ graph LR
+   subgraph left[" "]
+      direction TB
+      P10[Parameter 1]-->S0["Relu"]
+      S0-->Add0[Add]
+      P20[Parameter 2]--->Add0-->R0["Result"]
+   end
+   subgraph middle[" "]
+      direction TB
+      P11[Parameter 1]-->S1["Subgraph"] 
+      S1-->Add1[Add]
+      P21[Parameter 2]--->Add1-->R1["Result"]
+   end
+   subgraph right[" "]
+      direction TB
+      P12[Parameter 1]--->S2["Subgraph"]
+      P22[Parameter 2]--->S2-->R2["Result"]
+   end
+   left-->|<font size=+1>Start Subgraph\n with Relu </font>|middle
+   middle-->|<font size=+1>Attach Add\n to Subgraph</font>|right
+   classDef no-bg-color fill:none,stroke-width:0px
+   class left,middle,right no-bg-color
 ```
 
-Broadcast and regular streaming vector load is possible from the same pointer. BroadcastLoad should always go before streaming load. BroadcastLoad for non the most varying dimension is not generated, however it affects the generated schedule.
-
-#### Target-specific optimizations
+The algorithm described above reflects a general idea behind the tokenization process, but there are of course other details that should be considered in a real-life implementation. For instance, the `Tokenizer` should take additional care to avoid creation of cyclic dependencies when attaching a Node to `Subgraph`. Similarly, `Subgraph` is not allowed to connect with more than one `Result` node (no node can do that). In both of the cases, a new `Subgraph` is started instead of attaching to the existing one to avoid an illegal configuration. Another interesting case is the tokenization of constant nodes. If a `Constant` represents a single value (scalar) and its child was attached to a `Subgraph`, then the `Constant` also attached to the same `Subgraph`. If a `Constant` is not scalar, then it can't be tokenized since storing `Constant's` values requires memory management, and memory management is a plugin responsibility. Please refer to the [collapse_subgraph.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/pass/collapse_subgraph.cpp) to gain more insights on the tokenization process.
```

Comment:
`Subgraph is not allowed to connect with more than one Result node (no node can do that)`
I'm not sure that it's good explanation for `Subgraph` :smile:
Because some operations can have several outputs (`MaxPool-8`, for example). For snippets, it's a problem because origin model expects `Results` with special names. But `Subgraph` cannot have several names. So if we fuse ops before `Result` into `Subgraph`, we cannot find the `Result` by name because `Subgraph` has only one name. Moreover, as I remember, outputs are described by tensor names. I think because of that simple `nGraph` nodes can have several outputs

Reply:
Also I'd suggest to create the new paragraph with name `limitations` where you can describe limitations of tokenization (count of `Result` nodes, cycle dependencies, the plugin limitation with gpr count - need to notice that it's plugin specific limitation and we're going to remove it - ticket is in code)

Reply:
The ticket for a more detailed Tokenization description: 111891